src/util/GuardedStream.ts

```diff
@@ -74,7 +76,13 @@ function emitStoredErrors(this: Guarded, event: string): void {
 /**
  * Makes sure that listeners always receive the error event of a stream,
  * even if it was thrown before the listener was attached.
- * If the input is already guarded nothing will happen.
+ *
+ * When guarding a stream it is assumed that error listeners already attached should be ignored,
+ * only error listeners attached after the stream is guarded will prevent an error from being logged.
+ *
+ * If the input is already guarded nothing will happen the guard will be reset,
```

Comment:
Something wrong with this sentence?

Reply:
Ah yea, some words in the wrong order there.